lib/validators/specResolver.js

```diff
@@ -488,6 +501,281 @@ class SpecResolver {
     }
     return Promise.resolve(self);
   }
+
+  /**
+ * Deletes all the references to allOf from all the model definitions in the swagger spec.
+ */
+  resolveDiscriminator() {
+    let self = this;
+    let spec = self.specInJson;
+    let definitions = spec.definitions;
+    let modelNames = Object.keys(definitions);
+    let subTreeMap = new Map();
+    let references = JsonRefs.findRefs(spec);
+
+    modelNames.map((modelName) => {
+      if (definitions[modelName].discriminator) {
+        let modelReference = `#/definitions/${modelName}`;
+        let root = subTreeMap.get(modelName), locationsToBeUpdated = [];
+        if (!root) {
+          root = this.createPolymorphicTree(modelName, definitions[modelName].discriminator, subTreeMap);
+        }
+        for (let key in references) {
+          if (references[key].uri === modelReference && key.indexOf("allOf") === -1) locationsToBeUpdated.push(key);
+        }
+        for (let location of locationsToBeUpdated) {
+          utils.setObject(spec, `${location.slice(1)}`, { "oneOf": this.buildOneOfReferences(root) });
```

Comment:
This statement removes the description property along with the $ref that we actually want to remove. Hence, performing the following steps would be a better approach:
- doing a `utils.getObject()`
- deleting the `$ref` property
- adding the `oneOf` array of references comprising of the parent and the child
- doing a `utils.setObject()` with the updated object

```javascript
for (let location of locationsToBeUpdated) {
  const slicedLocation = location.slice(1);
  let obj = utils.getObject(spec, slicedLocation);
  if (obj) {
    if (obj['$ref']) delete obj['$ref'];
    obj.oneOf = this.buildOneOfReferences(root);
    utils.setObject(spec, slicedLocation, obj);
  }
}
```

Reply:
👍